src/FakeItEasy/TaskHelper.cs

```diff
@@ -1,14 +1,59 @@
 namespace FakeItEasy
 {
+    using System;
+    using System.Collections.Concurrent;
+    using System.Reflection;
     using System.Threading.Tasks;
 
     internal static class TaskHelper
     {
+        private static readonly Lazy<Task> CachedCancelledTask = new Lazy<Task>(CreateGenericCancelledTask<int>);
+
+        private static readonly ConcurrentDictionary<Type, Task> CachedCancelledTasks = new ConcurrentDictionary<Type, Task>();
+
+        private static readonly MethodInfo CreateGenericCancelledTaskGenericDefinition =
+            typeof(TaskHelper).GetMethod(
+                nameof(CreateGenericCancelledTask),
+                BindingFlags.Static | BindingFlags.NonPublic);
+
         public static Task<T> FromResult<T>(T result)
         {
             var source = new TaskCompletionSource<T>();
             source.SetResult(result);
             return source.Task;
         }
+
+        public static Task Cancelled()
+        {
+            return CachedCancelledTask.Value;
+        }
+
+        public static Task Cancelled(Type resultType)
+        {
+            if (resultType == typeof(void))
+            {
+                return Cancelled();
+            }
+
+            return CachedCancelledTasks.GetOrAdd(
+                resultType,
+                type =>
+                {
+                    var method = CreateGenericCancelledTaskGenericDefinition.MakeGenericMethod(type);
+                    return (Task)method.Invoke(null, new object[0]);
+                });
+        }
+
+        public static Task Cancelled<T>()
+        {
+            return Cancelled(typeof(T));
+        }
+
+        private static Task<T> CreateGenericCancelledTask<T>()
+        {
+            var source = new TaskCompletionSource<T>();
+            source.TrySetCanceled();

```

Comment:
You're right, `SetCanceled` has no reason to fail here. I was under the impression that `TrySetCanceled` was the "right" way to do it, because they added an overload with a `CancellationToken` in 4.6 for `TrySetCanceled` but not for `SetCanceled`. It doesn't matter in this case though, I'll change it to `SetCancelled`.